app/views/hub/users/index.html.erb

```diff
@@ -65,3 +70,47 @@
     </table>
   </div>
 <% end %>
+
+<% content_for :script do %>
+  <script>
+      document.addEventListener("DOMContentLoaded", function () {
+          const button = document.getElementById("toggle-suspended-btn");
+          let hidden = localStorage.getItem("hideSuspended") === "true";
+
+          function applyFilter() {
+              document.querySelectorAll(".user-row").forEach(row => {
+                  if (row.dataset.suspended === "true") {
+                      row.style.display = hidden ? "none" : "";
+                  }
+              });
+
+              button.textContent = hidden ? "Show Suspended Users" : "Hide Suspended Users";
+              button.style.backgroundColor = hidden ? "black" : "white";
+              button.style.color = hidden ? "white" : "black";
+          }
+
+          applyFilter();
+
+          button.addEventListener("click", function () {
+              hidden = !hidden;
+              localStorage.setItem("hideSuspended", hidden);
+              applyFilter();
+          });
+
+          const observer = new MutationObserver(mutations => {
```

Comment:
This is neat, I didn't know about MutationObserver. Do we actually mutate the contents of the `#users-table` while the page is loaded?